core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorUtil.java

```diff
@@ -1039,6 +1046,25 @@ public static ImmutableList<ImmutableBitSet> cube(
     return schema.getType(name, false);
   }
 
+  // public static SqlIdentifier getNameFromType(
+  //     CalciteSchema rootSchema, RelDataType type) {
+  //   // SqlIdentifier target = type.;
+  //   Queue<CalciteSchema> queue = new ArrayDeque<>();
+  //   queue.add(rootSchema);
+  //   while(!queue.isEmpty()){
+  //     CalciteSchema currentSchema = queue.remove();
+  //     if currentSchema.getType
+  //   }
+  //   return null;
+  // }
+  public static String getAliasedTypeName(SqlValidator validator, String typeName){
```

Comment:
this is another important function where given a Calcite type we return the dialects reverse aliased type 
(i.e. calcite TIMESTAMP -> bigquery DATETIME)